### ISSUE TITLE
Add support for Python 3.11

### ISSUE THREAD

**Issue by pmeier** [pytorch/vision#7049]
As of today, Python 3.11 is not available in the `defaults` channel of Anaconda, which we are building against: https://anaconda.org/anaconda/python

As stated in https://github.com/pytorch/vision/pull/6841#issuecomment-1294410859, we cannot build against the `conda-forge` channel. 

As soon as it is available in the `defaults` channel, there is #6959 to add support in our CI and in turn for the nightlies and the next release.

**Comment by CtrlC-Root** [comment]
https://github.com/ContinuumIO/anaconda-issues/issues/13082

**Comment by johnnynunez** [comment]
```conda create --name m3 python=3.11 -c conda-forge -y```

**Comment by pmeier** [comment]
@johnnynunez You are using the `conda-forge` channel. As explained above, we are not building against that, but rather the `defaults` channel:

```
$ conda create --name pytorch python=3.11 --dry-run
[...]
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - python=3.11

Current channels:

  - https://repo.anaconda.com/pkgs/main/linux-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/linux-64
  - https://repo.anaconda.com/pkgs/r/noarch
[...]

**Comment by johnnynunez** [comment]
> @johnnynunez You are using the `conda-forge` channel. As explained above, we are not building against that, but rather the `defaults` channel:
> 
> ```
> $ conda create --name pytorch python=3.11 --dry-run
> [...]
> Solving environment: failed
> 
> PackagesNotFoundError: The following packages are not available from current channels:
> 
>   - python=3.11
> 
> Current channels:
> 
>   - https://repo.anaconda.com/pkgs/main/linux-64
>   - https://repo.anaconda.com/pkgs/main/noarch
>   - https://repo.anaconda.com/pkgs/r/linux-64
>   - https://repo.anaconda.com/pkgs/r/noarch
> [...]
> ```

They are working on that: https://github.com/conda/conda/issues/11170
Why does it rely on conda and not on python binaries as in pytorch repository?

**Comment by pmeier** [comment]
> Why does it rely on conda and not on python binaries as in pytorch repository?

Because we are not just building against the Python ABI, but a few other libraries as well. Mainly `libpng` and `libjpeg`.

**Comment by johnnynunez** [comment]
> > Why does it rely on conda and not on python binaries as in pytorch repository?
> 
> Because we are not just building against the Python ABI, but a few other libraries as well. Mainly `libpng` and `libjpeg`.

Thank you for clarify

**Comment by pmeier** [comment]
@malfet I favor this. If we do, we should also consider 4. of #6563 since we are missing out on quite a big performance improvement again due to lack of support in the defaults channel.

**Comment by johnnynunez** [comment]
@pmeier Python 3.11 is out in anaconda
https://github.com/ContinuumIO/anaconda-issues/issues/13082#issuecomment-1386874942

**Comment by pentium10** [comment]
any progress on this one? pytorch was enabled for linux builds, and wondering if vision is up to a point to be tested?

**Comment by pmeier** [comment]
@pentium10 As explained above, we need more than just `python=3.11` to be available. See https://github.com/pytorch/vision/pull/7108#issuecomment-1399901527. Until all of that is available as well, we can't provide 3.11 builds.

**Comment by pentium10** [comment]
numpy, Pillow, scipy, requests, pytest, setuptools all have release builds for python3.11
what do I miss?

**Comment by pmeier** [comment]
@pentium10 You miss that they have no builds against Python 3.11 on the conda defaults channel. And that is what we are building against. So we need Anaconda to build these packages for 3.11 now that they recently added support for it. But there is nothing that we can do on that front other than ask.

**Comment by malfet** [comment]
@pmeier I agree we could not publish conda package until dependencies are not available on Anaconda, but I somewhat disagree that we should hold of wheel builds due to that. 
But to unblock some of the pipelines I'm going to build and publish some of the dependencies to `pytorch-nightly` channel (already published `pytest`, `numpy`, `scipy`  and `setuptools` for Linux, can easily add `pillow` there)

**Comment by powderluv** [comment]
yes can we please have 3.11 torchvision wheels

**Comment by cornking03** [comment]
> > Why does it rely on conda and not on python binaries as in pytorch repository?
> 
> Because we are not just building against the Python ABI, but a few other libraries as well. Mainly `libpng` and `libjpeg`.

Is there a document/post explaining (in terms that someone who does not develop libraries can understand) the reliance on Anaconda for the CI process? The `default` channel [may require a paid license](https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/channels.html#:~:text=The%20conda%20command%20searches%20a,free%20for%20all%20to%20use.), so I was just curious why that was chosen. 

Are there any alternatives to Anaconda for CI? This seems to be one of the big things hindering progress. I'm not trying to suggest that retooling a CI process would be trivial - just trying to understand the design decisions.

**Comment by malfet** [comment]
> yes can we please have 3.11 torchvision wheels

I think we do have them now.

**Comment by powderluv** [comment]
just validated it works in downstream torch-mlir CI https://github.com/llvm/torch-mlir/pull/1848

Thank you.

**Comment by Erotemic** [comment]
Any update for 3.11 torchvision on pypi?

**Comment by pmeier** [comment]
@Erotemic Release is planned for March 15.
